Puzzle.py

```diff
@@ -0,0 +1,269 @@
+import numpy as np
+import pygame
+import PIL
+import random
+
+'''
+Parent class Puzzle
+    Description:
+        All image puzzles are a subset of this class.
+        This class splits a puzzle image up into individual pieces that can be modified, moved, and deleted.
+        It is generally assumed that the puzzle is square, so rectangular puzzles will have rectangular pieces.
```

Comment:
I don't think this assumption is necessary for doing things (though I haven't looked at the code yet...)

Reply:
I changed it. I was referring to the number of pieces wide vs the number of pieces tall.